### PR TITLE
[FIX] Update setup and requirements

### PR DESCRIPTION
**Pull request by r0mainK**
@vmarkovtsev tests are failing in my PR in apollo because `sourced-ml` isn't installed using requirements  but using `setup.py`, which is not up to data. Also, `tensorflow` seems to no longer be an option, as `keras` uses it as backend (initially I got the following error log, although when adding tensorflow to travis I got problems bc `pandas`, `scikit-learn` or `tqdm` was  missing):

EDIT: saw in the other pending PR you don't want to require tensorflow, not sure how you wanna circumvent since it's used extensively in a lot of `id_splitter`

```
 File "/home/travis/build/src-d/apollo/apollo/bags.py", line 2, in <module>
    from sourced.ml.cmd import repos2bow_template, repos2bow_index
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/sourced/ml/cmd/__init__.py", line 17, in <module>
    from sourced.ml.cmd.train_id_split import train_id_split
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/sourced/ml/cmd/train_id_split.py", line 6, in <module>
    from sourced.ml.algorithms.id_splitter import set_random_seed, config_keras, \
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/sourced/ml/algorithms/id_splitter/__init__.py", line 2, in <module>
    from sourced.ml.algorithms.id_splitter.features import prepare_features, read_identifiers
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/sourced/ml/algorithms/id_splitter/features.py", line 7, in <module>
    from keras.preprocessing.sequence import pad_sequences
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/keras/__init__.py", line 3, in <module>
    from . import utils
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/keras/utils/__init__.py", line 6, in <module>
    from . import conv_utils
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/keras/utils/conv_utils.py", line 9, in <module>
    from .. import backend as K
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/keras/backend/__init__.py", line 89, in <module>
    from .tensorflow_backend import *
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/keras/backend/tensorflow_backend.py", line 5, in <module>
    import tensorflow as tf
ImportError: No module named 'tensorflow'
```

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -11,3 +11,4 @@ docker>=3.4.0,<4.0
 pandas>=0.22,<1.0
 scikit-learn>=0.19,<1.0
 tqdm>=4.20,<5.0
+tensorflow>=1.0,<2.0
```
Comment (vmarkovtsev) [inline requirements.txt]: So we do not require Tensorflow because there are two versions: gpu and non-gpu and they are in conflict.

Comment (vmarkovtsev) [reply requirements.txt]: CI installs it separately.

File: setup.py [hunk 2]
```diff
@@ -30,13 +30,17 @@
...
                       "pygments>=2.2.0,<3.0",
-                      "keras>=2.0,<3.0"] + typing,
+                      "keras>=2.0,<3.0",
+                      "docker>=3.4.0,<4.0",
+                      "pandas>=0.22,<1.0",
+                      "scikit-learn>=0.19,<1.0",
+                      "tqdm>=4.20,<5.0",
+                      "tensorflow>=1.0,<2.0"] + typing,
     extras_require={
-        "tf": ["tensorflow>=1.0,<2.0"],
```
Comment (vmarkovtsev) [inline setup.py]: You install it like this:
```
pip3 install sourced-ml[tf]
```
or
```
pip3 install sourced-ml[tf-gpu]
```

File: setup.py [hunk 3]
```diff
@@ -30,13 +30,17 @@
...
                       "modelforge>=0.7.0,<0.8",
-                      "sourced-engine>=0.5.1,<0.7",
-                      "humanize>=0.5.0",
+                      "sourced-engine>=0.7.0,<0.8",
+                      "humanize>=0.5.0,<0.6",
                       "parquet>=1.2,<2.0",
                       "pygments>=2.2.0,<3.0",
-                      "keras>=2.0,<3.0"] + typing,
+                      "keras>=2.0,<3.0",
+                      "docker>=3.4.0,<4.0",
```
Comment (vmarkovtsev) [inline setup.py]: We already have docker in `tests-require`

File: setup.py [hunk 4]
```diff
@@ -30,13 +30,17 @@
...
+                      "pandas>=0.22,<1.0",
+                      "scikit-learn>=0.19,<1.0",
```
Comment (vmarkovtsev) [inline setup.py]: Do we really use sklearn inside?

Comment (r0mainK) [reply setup.py]: Yeah was surprised, in `cmd/id2role_eval.py`

File: setup.py (hunk 2 above)
Comment (r0mainK) [reply setup.py]: Okay, do you mind if I add it to docs regarding setup ?

File: setup.py (hunk 3 above)
Comment (r0mainK) [reply setup.py]: Removing then

File: setup.py (hunk 2 above)
Comment (vmarkovtsev) [reply setup.py]: sure

**Pull request merged**